lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -317,26 +315,33 @@ def add_payment_method(xml, amount, payment_method, options)
       end
 
       def add_three_d_secure(three_d_secure, xml)
-        xml.tag! 'info3DSecure' do
-          xml.tag! 'threeDSVersion', three_d_secure[:version]
+        xml.info3DSecure do
+          xml.threeDSVersion three_d_secure[:version]
           if three_d_secure[:version] =~ /^2/
-            xml.tag! 'dsTransactionId', three_d_secure[:ds_transaction_id]
+            xml.dsTransactionId three_d_secure[:ds_transaction_id]
           else
-            xml.tag! 'xid', three_d_secure[:xid]
+            xml.xid three_d_secure[:xid]
           end
-          xml.tag! 'cavv', three_d_secure[:cavv]
-          xml.tag! 'eci', three_d_secure[:eci]
+          xml.cavv three_d_secure[:cavv]
+          xml.eci three_d_secure[:eci]
         end
       end
 
       def add_card(xml, payment_method, options)
-        xml.tag! 'cardNumber', payment_method.number
-        xml.tag! 'expiryDate' do
-          xml.tag! 'date', 'month' => format(payment_method.month, :two_digits), 'year' => format(payment_method.year, :four_digits)
+        xml.cardNumber payment_method.number
+        xml.expiryDate do
+          xml.date(
+            'month' => format(payment_method.month, :two_digits),
+            'year' => format(payment_method.year, :four_digits)
+          )
         end
 
-        xml.tag! 'cardHolderName', options[:execute_threed] && !options[:three_ds_version]&.start_with?('2') ? '3D' : payment_method.name
-        xml.tag! 'cvc', payment_method.verification_value
+        if options[:execute_threed] && !options[:three_ds_version]&.start_with?('2')
```

Comment:
This was another weird RuboCop case where it would complain that the complicated logic needed to be surrounded by parentheses but then when I added them, it would complain that you shouldn't enclose a ternary condition with parentheses... This seemed like a reasonable way to appease the RuboCop overlords.

Reply:
Would this work? It seems to work on my laptop (and RuboCop isn't screaming about it):
```
xml.cardHolderName(options[:execute_threed] && !options[:three_ds_version]&.start_with?('2') ? '3D' : payment_method.name)
```
Although it might be more readable to break it up like maybe:
```
card_holder_name = options[:execute_threed] && !options[:three_ds_version]&.start_with?('2') ? '3D' : payment_method.name
xml.cardHolderName(card_holder_name)
```

Reply:
That does look much better, especially breaking it out to two lines! Thanks for the suggestion!